string[] good_bins
string[] bad_bins
---
int32 result
---
